packages/oauth/oauth-provider/src/errors/invalid-client-metadata-error.ts

```diff
@@ -12,8 +12,16 @@ export class InvalidClientMetadataError extends OAuthError {
     super('invalid_client_metadata', error_description, 400, cause)
   }
 
-  static from(cause: unknown): InvalidClientMetadataError {
-    if (cause instanceof InvalidClientMetadataError) return cause
-    return new InvalidClientMetadataError('Invalid client configuration', cause)
+  static from(
+    cause: unknown,
+    fallbackMessage = 'Invalid client metadata document',
+  ): OAuthError {
+    if (cause instanceof OAuthError) {
+      return cause
+    }
+    if (cause instanceof TypeError) {
+      return new InvalidClientMetadataError(cause.message, cause)
+    }
```

Comment:
I don't think we need to change this, but it is a bit interesting that a system `TypeError` (e.g. "null has no properties") could make it into a user-facing error, if I'm understanding correctly.

Reply:
This is due to the fact that this error is meant to be used in the context of parsing a client_id, meaning that any TypeError (e.g. InvalidURL) is actually not a code error but rather a user data error. I could move that logic elsewhere where `TypeError` are actually thrown ?

Reply:
I added a comment to explain the reasonning.